src/VPolytope.jl

```diff
@@ -496,15 +496,22 @@ Return an `VRep` polyhedron from `Polyhedra.jl` given a polytope in V-representa
 - `backend` -- (optional, default: `default_polyhedra_backend(P, N)`) the polyhedral
                computations backend, see [Polyhedra's documentation](https://juliapolyhedra.github.io/Polyhedra.jl/latest/installation.html#Getting-Libraries-1)
                for further information
+- `ambient_dimension` -- (default, optional: `dim(P)`) ambient dimension of the
+                         polytope; required if the polytope is empty
 
 ### Output
 
 A `VRep` polyhedron.
 """
 function polyhedron(P::VPolytope{N};
-                    backend=default_polyhedra_backend(P, N)) where {N<:Real}
-    V = hcat(vertices_list(P)...)'
-    return polyhedron(Polyhedra.vrep(V), backend)
+                    backend=default_polyhedra_backend(P, N),
+                    ambient_dimension::Int=dim(P)) where {N<:Real}
+    if ambient_dimension == -1 # empty polytope
```

Comment:
This error message is not shown because `-1` is not the default value. Instead the method call will crash when it tries to execute `dim(P)`. Instead I suggest the following:
1. Remove the default value for `ambient_dimension`.
2. Only pass the dimension if the vertices list is empty (otherwise use the old implementation).
3. If the list is empty, test if `ambient_dimension` is defined. If not, throw the error you have here.

Reply:
`-1` is the default value for an empty `VPolytope`, see [dim(::Vpolytope)](https://github.com/JuliaReach/LazySets.jl/blob/master/src/VPolytope.jl#L74).

Reply:
I see :+1:

Reply:
Should we rather add a field for the dimension (see #1201 for a related discussion) or assert in the constructor that there is at least one vertex? It seems a bit strange to have this workaround and keep the problem on our side.

Reply:
I agree to add a new field for `EmptySet` as in [this comment](https://github.com/JuliaReach/LazySets.jl/issues/1201#issue-417734408). Are you suggesting to add a field for the dimension in `VPolytope`? We could but i dont think it is justified to handle this particular case (creating an empty set through a `VPolytope()`).  Besides, I don't think that the workaround of this PR poses problems on our side, does it?

OTOH i checked again the behavior of `Polyhedra` and calling this argument `ambient_dimension` is not consistent (compare to `LazySets` dim):

```julia
julia> p = polyhedron(vrep([rand(2) for _ in 1:2]))
Polyhedron DefaultPolyhedron{Float64,Polyhedra.Intersection{Float64,Array{Float64,1},Int64},Polyhedra.Hull{Float64,Array{Float64,1},Int64}}:
2-element iterator of Array{Float64,1}:
 [0.905471, 0.873191]
 [0.774891, 0.187635]

julia> dim(p)
1

julia> p = polyhedron(vrep(Vector{Vector{Float64}}(), d=2))
Polyhedron DefaultPolyhedron{Float64,Polyhedra.Intersection{Float64,Array{Float64,1},Int64},Polyhedra.Hull{Float64,Array{Float64,1},Int64}}

julia> dim(p)
-1
```

Reply:
> Are you suggesting to add a field for the dimension in VPolytope?

I would rather forbid this case in the constructor.
The problem I see is that we call this method internally, and obviously we do not pass this argument. So you fixed the one problem where the user called `polyhedron`, but all other cases cannot be handled, so you would have to add this keyword argument everywhere. To me this PR just hides the problem a bit deeper.

> ```julia
> julia> dim(p)
> 1
> ```

Huh, why is the dimension `1` here?

> is not consistent

Why?

Reply:
> Huh, why is the dimension 1 here?

Okay, I understood this part now with the discussion below.

I still think that this PR does not solve the problem but only hides it. But let us move on.

Reply:
👍  In Polyhedra.jl it is legit to create an empty polytope if you pass the kwarg `d`; this PR just adds this feature to LazySets.

Reply:
But only for manually creating the `Polyhedra.polyhedron`. If you work with an empty `LazySets.VPolytope` and apply an operation that internally converts to `Polyhedra.polyhedron`, it will still crash.

Reply:
Ok true, we should fix these when they appear (https://github.com/JuliaReach/LazySets.jl/pull/1371/ and https://github.com/JuliaReach/LazySets.jl/pull/1369 cover two such operations, that were breaking when the polytope is empty).

Reply:
But you can only fix them by requiring to pass this additional argument (that was my point [here](https://github.com/JuliaReach/LazySets.jl/pull/1373#discussion_r283502911).) And in the application you mention, I doubt that you know the relative dimension in advance.

Reply:
- In the case of `tohrep` of an empty `VPolytope`, again it is special-cased before the call to `polyhedron`, so we don't need to know its relative dimension.
- In the case of the `convex_hull`, before it was needed (because it was directly calling `Polyhedra.convex_hull`), but now it calls `LazySets.convex_hull` which operates on lists of vertices (and the empty case is handled separately) so we don't need to know the relative dimension.

I think this fixes the issues that have arised so far, but you are right, there may be more that we haven't identified yet.